src/python/pants/backend/jvm/tasks/jvm_compile/jvm_compile.py

```diff
@@ -283,19 +283,20 @@ def extra_resources(self, compile_context):
 
     return result
 
-  def extra_resources_digest(self, compile_context):
-    """Compute a Digest for the extra_resources for the given context."""
+  def extra_post_compile_resources_digest(self, compile_context):
```

Comment:
Should change the name of this method if it's going to have sideeffects like this (which is ok I think).

Reply:
Removed the side effects instead. It's still under `tmpdir`, but changed `<tmpdir>/<relative path to buildroot>/post_merge_compile_dir/META-INF/...` instead of `tmpdir/META-INF` directly.